### ISSUE TITLE
QAT 4-bit: Abnormally Large Loss (>100) When Fine-tuning Qwen3-1.7B on Wikitext

### ISSUE THREAD

**Issue by quanfeifan** [facebookresearch/ParetoQ#15]
I’m encountering abnormally large training loss values (often >8 ) when performing 4-bit Quantization-Aware Training (QAT) on the Qwen3-1.7B model using the Wikitext dataset (wikitext-103-raw-v1) for causal language modeling.
Could you please advise how to resolve this? Or what kind of dataset would help stabilize training? Previously, using the OpenWebText dataset resulted in even higher loss and gradient norm values. 

`{'loss': 8.5418, 'grad_norm': 28319940608.0, 'learning_rate': 2e-05, 'epoch': 0.0}      
                                                                            
{'loss': 8.3583, 'grad_norm': 12415139840.0, 'learning_rate': 1.9999999987976955e-05, 'epoch': 0.0}      
                                                           
{'loss': 8.4148, 'grad_norm': 54760833024.0, 'learning_rate': 1.9999999951907812e-05, 'epoch': 0.0}    
                                                             
{'loss': 8.4432, 'grad_norm': 231928233984.0, 'learning_rate': 1.999999989179258e-05, 'epoch': 0.0}      
                                                           
{'loss': 8.3209, 'grad_norm': 118111600640.0, 'learning_rate': 1.9999999807631257e-05, 'epoch': 0.0}       
                                                         
{'loss': 8.3343, 'grad_norm': 77846282240.0, 'learning_rate': 1.9999999699423838e-05, 'epoch': 0.0}       
                                                          
{'loss': 8.3259, 'grad_norm': 2382364672.0, 'learning_rate': 1.9999999567170326e-05, 'epoch': 0.0}          
                                                        
{'loss': 8.4059, 'grad_norm': 96099893248.0, 'learning_rate': 1.9999999410870725e-05, 'epoch': 0.0}`

**Comment by mikezhaocs** [comment]
The sample code is designed specifically for Llama models; you'll need to apply comparable modifications for other models. When using 4-bit quantization, the loss at step 0 should be very similar to that of the unquantized model.
